Kalau di aku dia error gara2 UA masihan mas:

```
(rojak) % scrapy crawl kompascom
2016-10-13 05:57:46 [scrapy] INFO: Scrapy 1.2.0 started (bot: rojak_pantau)
2016-10-13 05:57:46 [scrapy] INFO: Overridden settings: {'NEWSPIDER_MODULE': 'rojak_pantau.spiders', 'SPIDER_MODULES': ['rojak_pantau.spiders'], 'ROBOTSTXT_OBEY': True, 'USER_AGENT': 'Mozilla/5.0 (iPhone; CPU iPhone OS 9_1 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13B143 Safari/601.1', 'BOT_NAME': 'rojak_pantau'}
2016-10-13 05:57:46 [scrapy] INFO: Enabled extensions:
['scrapy.extensions.logstats.LogStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.corestats.CoreStats']
2016-10-13 05:57:46 [kompascom] INFO: Fetching media information
Unhandled error in Deferred:
2016-10-13 05:57:46 [twisted] CRITICAL: Unhandled error in Deferred:


Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/commands/crawl.py", line 57, in run
    self.crawler_process.crawl(spname, **opts.spargs)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/crawler.py", line 163, in crawl
    return self._crawl(crawler, *args, **kwargs)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/crawler.py", line 167, in _crawl
    d = crawler.crawl(*args, **kwargs)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1331, in unwindGenerator
    return _inlineCallbacks(None, gen, Deferred())
--- <exception caught here> ---
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1185, in _inlineCallbacks
    result = g.send(result)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/crawler.py", line 90, in crawl
    six.reraise(*exc_info)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/crawler.py", line 71, in crawl
    self.spider = self._create_spider(*args, **kwargs)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/crawler.py", line 94, in _create_spider
    return self.spidercls.from_crawler(self, *args, **kwargs)
  File "/home/vagrant/rojak/rojak-pantau/rojak_pantau/spiders/kompascom.py", line 67, in from_crawler
    *args, **kwargs)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/spiders/__init__.py", line 50, in from_crawler
    spider = cls(*args, **kwargs)
  File "/home/vagrant/rojak/rojak-pantau/rojak_pantau/spiders/kompascom.py", line 49, in __init__
    self.media['id'] = row[0]
exceptions.TypeError: 'NoneType' object has no attribute '__getitem__'
2016-10-13 05:57:46 [twisted] CRITICAL: 
Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1185, in _inlineCallbacks
    result = g.send(result)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/crawler.py", line 90, in crawl
    six.reraise(*exc_info)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/crawler.py", line 71, in crawl
    self.spider = self._create_spider(*args, **kwargs)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/crawler.py", line 94, in _create_spider
    return self.spidercls.from_crawler(self, *args, **kwargs)
  File "/home/vagrant/rojak/rojak-pantau/rojak_pantau/spiders/kompascom.py", line 67, in from_crawler
    *args, **kwargs)
  File "/home/vagrant/.virtualenvs/rojak/local/lib/python2.7/site-packages/scrapy/spiders/__init__.py", line 50, in from_crawler
    spider = cls(*args, **kwargs)
  File "/home/vagrant/rojak/rojak-pantau/rojak_pantau/spiders/kompascom.py", line 49, in __init__
    self.media['id'] = row[0]
TypeError: 'NoneType' object has no attribute '__getitem__'
```

Kalau news itemnya udah ke pipeline seharusnya udah bisa sih

--- a/rojak-pantau/rojak_pantau/spiders/kompascom.py
+++ b/rojak-pantau/rojak_pantau/spiders/kompascom.py
@@ -135,10 +135,10 @@ class KompasComSpider(scrapy.Spider):
         info_date = info_time[0].strip().split(' ');
         info_hours = info_time[1].strip().split(' ')[0].strip();
         day = info_date[0];
-        month = _.translate("idn")[info_date[1]];
+        month = _(info_date[1]);
         year = info_date[2];
         formatted_date = day+' '+month+' '+year+', ' + info_hours;
-        return datetime.strptime(formatted_date, '%d %B %Y, %H:%M');
+        return  wib_to_utc(datetime.strptime(formatted_date, '%d %B %Y, %H:%M'));
 
     def parse_news(self, response):
         self.logger.info('parse_news: %s' % response)

--- a/rojak-pantau/rojak_pantau/spiders/kompascom.py
+++ b/rojak-pantau/rojak_pantau/spiders/kompascom.py
@@ -28,7 +28,7 @@ UPDATE `media` SET last_scraped_at=UTC_TIMESTAMP() WHERE name=%s;
 '''
 
 class KompasComSpider(scrapy.Spider):
-    name = "kompascom"
+    name = "kompas"
     start_urls = [
         'http://lipsus.kompas.com/topikpilihanlist/3754/1/Pilkada.DKI.2017'
     ]

--- a/rojak-pantau/rojak_pantau/spiders/kompascom.py
+++ b/rojak-pantau/rojak_pantau/spiders/kompascom.py
@@ -24,7 +24,7 @@ SELECT id,last_scraped_at FROM media WHERE name=%s;
 '''
 
 sql_update_media = '''
-UPDATE `media` SET last_scraped_at=%s WHERE name=%s;
+UPDATE `media` SET last_scraped_at=UTC_TIMESTAMP() WHERE name=%s;
 '''
 
 class KompasComSpider(scrapy.Spider):
@@ -32,6 +32,10 @@ class KompasComSpider(scrapy.Spider):
     start_urls = [
         'http://lipsus.kompas.com/topikpilihanlist/3754/1/Pilkada.DKI.2017'
     ]
+
+    custom_settings = {
+        'USER_AGENT': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0'
+    }
 
     # Initialize database connection then retrieve media ID and
     # last_scraped_at information
@@ -82,7 +86,7 @@ class KompasComSpider(scrapy.Spider):
         if reason == 'finished':
             try:
                 self.logger.info('Updating media last_scraped_at information')
-                self.cursor.execute(sql_update_media, [datetime.now(), self.name])
+                self.cursor.execute(sql_update_media, [self.name])
                 self.db.commit()
                 self.db.close()
             except mysql.Error as err:

